### ISSUE TITLE
WCAG 2.2: Fix focus isn't moved to certain screens when opened (e.g. focus is lost after opening the note history screen).

### ISSUE THREAD

**Issue by pedr** [laurent22/joplin#11648]
In the Note revision/history screen, it seems to be broken (focus seems to go back to notebook list selector).

In the following pages it seems like the focus is on the new screen, but I had to use Up/Down so `orca` had the chance to read the content, which I'm not sure if it is the right behaviour or not (do you have an opinion @personalizedrefrigerator ?):

- [x] Joplin Cloud Sycnh
- [x] Dropbox Synch
- [x] OneDrive Synch

**Comment by personalizedrefrigerator** [comment]
Note: This issue was originally added in part due to [this part of F85](https://www.w3.org/WAI/WCAG22/Techniques/failures/F85.html):
> Example 2: Setting focus to the document after dismissing a menu embedded on the page
>
> When a menu is dismissed, it is removed or hidden from the web page and focus is set to the document. The user must tab from the beginning of the navigation sequence to reach the point from which the menu was opened.

However, the note history screen isn't a menu (it's a separate screen), so changing how focus is managed might not be necessary here. **Edit**: Checking again, the note history page isn't a separate screen &mdash; it's displayed as part of the editor.

**Comment by personalizedrefrigerator** [comment]
> In the Note revision/history screen, it seems to be broken (focus seems to go back to notebook list selector).

Good catch! The note history screen does seem to be broken. The focus seems to jump to the first item in the document.

> In the following pages it seems like the focus is on the new screen, but I had to use Up/Down so orca had the chance to read the content, which I'm not sure if it is the right behaviour or not (do you have an opinion @personalizedrefrigerator ?):

Ideally `orca` would provide some information about the new page. If `<h1>`/`<h2>`/etc aren't present on these pages, adding them might help with this.

**Comment by pedr** [comment]
Thanks for the help, I will create a PR for these changes.

**Comment by pedr** [comment]
I could not find a way to prevent focus going to start of the document when the revision is rendered. I'm letting this issue open for now.

**Comment by personalizedrefrigerator** [comment]
> I could not find a way to prevent focus going to start of the document when the revision is rendered. I'm letting this issue open for now.

One option might be to show the revision viewer in a `<Dialog>` component, rather than by replacing the note editor. The `Dialog` component internally uses a [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog), which should handle autofocus and restoring focus when the dialog is closed.

**Comment by pedr** [comment]
> One option might be to show the revision viewer in a `<Dialog>` component, rather than by replacing the note editor. The `Dialog` component internally uses a [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog), which should handle autofocus and restoring focus when the dialog is closed.

Good idea, I'm going to do this. Thanks!

**Comment by pedr** [comment]
Fixed by https://github.com/laurent22/joplin/pull/11769
